snuba/migrations/migration.py

```diff
@@ -51,3 +53,12 @@ def forwards_local(self) -> Sequence[Operation]:
     @abstractmethod
     def backwards_local(self) -> Sequence[Operation]:
         raise NotImplementedError
+
+    def forwards(self, context: Context) -> None:
+        migration_id, logger, update_status = context
+        logger.info(f"Running migration: {migration_id}")
```

Comment:
It seems these logging calls could easily be in the around the forwards method. You can log after updating the status. This means you would not need to pass the logger and the migration id to the migration object that seems quite ad hoc.

Reply:
That's true in this particular case but I wanted to include it here because I don't think it's avoidable forever. I think it's likely we'll want to log something in between each operation on each node in future especially if there's an error.

Reply:
Ok, then I would advise to organize the code this way to avoid giving the improession that this class and the special case for 0001 are actually doing the same thing, since in python we do not have final classes.

1) Have an abstract migration interface with one method only: `forward` (and backward at a point I guess) that is abstract as well.
2) Extend that with MultiStepMigration that introduces the forwards_local and backwards_local. This class provides this `forwards` method.
3) multiple steps migrations will extend this one
4) the special case extends directly the parent abstract class and redefines its own `forwards` method. This is a single step migration so it does not have steps.

How does that sounds ?

Reply:
That sounds good to me, I renamed the new class to `CustomMigration` to try to make the intended usage more obvious.